docs/addons/writing-addons.md

```diff
@@ -0,0 +1,250 @@
+</div> 
+
+Going over the code snippet in more detail. When Storybook starts up:
+
+- It [registers](./addons-api.md#addonsregister) the addon.
+- [Adds](./addons-api.md#addonsadd) a new `panel` titled `My Addon` to the UI.
+- When selected, the `panel` renders the static `div` content.
+
+
+### Register the addon
+
+Finally, let’s hook it all up. Change `.storybook/main.js` to the following:
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+    'common/my-addon-storybook-registration.js.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+<div class="aside">
+When you register a Storybook addon, it will look for either <code>register.js</code> or <code>preset.js</code> as the entry points.
+</div>
+
+Run `yarn storybook` and you should see something similar to:
+
+![Storybook addon initial state](./addon-initial-state.png)
+
+
+### Display story parameter
+
+Next, let’s replace the `MyPanel` component from above to show the parameter.
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+    'common/storybook-addon-change-panel.js.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+The new version is made smarter by [`useParameter`](./addons-api.md#useparameter), which is a [React hook](https://reactjs.org/docs/hooks-intro.html) that updates the parameter value and re-renders the panel every time the story changes.
+
+The [addon API](./addons-api.md) provides hooks like this so all of that communication can happen behind the scenes. That means you can focus on your addon's functionality.
+
+
+### Using the addon with a story
+
+When Storybook was initialized it provided a small set of examples stories. Change your `Button.stories.js` to the following:
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+    'react/button-story-with-addon-example.js.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+
+After applying the changes to the story your Storybook UI will yield the following:
+
+<video autoPlay muted playsInline loop>
+  <source
+    src="addon-final-stage-optimized.mp4"
+    type="video/mp4"
+  />
+</video>
+
+### Root level preset.js
+
+Before publishing the addon, we'll need to make one final change. In the root directory of the addon create a new file called `preset.js` and add the following:
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+    'common/my-addon-root-level-preset.js.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+Making this change helps Storybook's addon registration. By design Storybook looks for either a `preset.js` or a `register.js` file located at the root level. Omitting this file would require additional configuration from the addon consumer.
+
+### Packaging and publishing
+
+With the example above we showed how to create a bare-bones addon. Before publishing and distributing your own addon make sure you have met the following requirements:
+
+- A `package.json file` with metadata about the addon.
```

Comment:
- Remove periods (these aren't sentences)
- No need to repeat "A..." for every item